### ISSUE TITLE
Node placed over edge in layout output

### ISSUE THREAD

**Issue by brollb** [OpenKieler/elkjs#42]
**tl;dr** - I am getting an undesirable layout from a small graph in which an edge is covered by a node. The easiest way to see it would be to copy the JSON into https://rtsys.informatik.uni-kiel.de/elklive/json.html. I have a suspicion that this may be better suited for the main ELK project and not elkjs (or maybe this is the desired behavior - although I hope not). Regardless, I will try to explain the situation below and any help is appreciated.

I have a simple graph with 4 nodes with basically two paths from the source to the terminal: 

    a -> b -> c -> d
    a -> b -> d

I expect it to be rendered in a way in which I can see all the connections and edges. However, the resulting layout shows essentially a path from **a** to **d** with the edge from **b** to **d** completely covered by **c**.

```
{
  "id": "root",
  "layoutOptions": {
    "elk.algorithm": "layered",
    "org.eclipse.elk.direction": "DOWN",
    "org.eclipse.elk.spacing.nodeNode": 40,
    "org.eclipse.elk.layered.spacing.nodeNodeBetweenLayers": 40
  },
  "edges": [
    {
      "id": "/x/H/k-/x/i/i",
      "source": "/x/H",
      "target": "/x/i",
      "sourcePort": "/x/H/k",
      "targetPort": "/x/i/i"
    },
    {
      "id": "/x/i/R-/x/W/f",
      "source": "/x/i",
      "target": "/x/W",
      "sourcePort": "/x/i/R",
      "targetPort": "/x/W/f"
    },
    {
      "id": "/x/i/R-/x/C/4",
      "source": "/x/i",
      "target": "/x/C",
      "sourcePort": "/x/i/R",
      "targetPort": "/x/C/4"
    },
    {
      "id": "/x/W/D-/x/C/4",
      "source": "/x/W",
      "target": "/x/C",
      "sourcePort": "/x/W/D",
      "targetPort": "/x/C/4"
    }
  ],
  "children": [
    {
      "id": "/x/H",
      "height": 50,
      "width": 100,
      "ports": [
        {
          "id": "/x/H/k",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "SOUTH"
          },
          "x": 50,
          "y": 74
        }
      ],
      "properties": {
        "org.eclipse.elk.portConstraints": "FIXED_POS"
      }
    },
    {
      "id": "/x/i",
      "height": 50,
      "width": 100,
      "ports": [
        {
          "id": "/x/i/i",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "NORTH"
          },
          "x": 50,
          "y": 24
        },
        {
          "id": "/x/i/R",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "SOUTH"
          },
          "x": 50,
          "y": 74
        }
      ],
      "properties": {
        "org.eclipse.elk.portConstraints": "FIXED_POS"
      }
    },
    {
      "id": "/x/W",
      "height": 50,
      "width": 164.90625,
      "ports": [
        {
          "id": "/x/W/f",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "NORTH"
          },
          "x": 82.453125,
          "y": 24
        },
        {
          "id": "/x/W/D",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "SOUTH"
          },
          "x": 82.453125,
          "y": 74
        }
      ],
      "properties": {
        "org.eclipse.elk.portConstraints": "FIXED_POS"
      }
    },
    {
      "id": "/x/C",
      "height": 50,
      "width": 122.828125,
      "ports": [
        {
          "id": "/x/C/4",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "NORTH"
          },
          "x": 61.4140625,
          "y": 24
        },
        {
          "id": "/x/C/r",
          "width": 1,
          "height": 1,
          "properties": {
            "org.eclipse.elk.port.side": "SOUTH"
          },
          "x": 61.4140625,
          "y": 74
        }
      ],
      "properties": {
        "org.eclipse.elk.portConstraints": "FIXED_POS"
      }
    }
  ]
}
```

**Comment by le-cds** [comment]
I've tried to reproduce the problem with the current ELK Nightly, but the results look fine:

![issue42](https://user-images.githubusercontent.com/7465584/43774753-8338d590-9a4a-11e8-8f21-5441bb05b9cc.png)

I'm guessing it might have been a bug that we have fixed since the last elk.js release was produced.

**Comment by brollb** [comment]
Ok, I will check out rebuilding the elkjs from the latest ELK and see if that fixes it for me! Thanks!

**Comment by uruuru** [comment]
Let me know if #41 is not an issue for you.

**Comment by brollb** [comment]
Haha, that's the issue I am currently fighting... I should've checked here before searching for solutions elsewhere.

**Comment by uruuru** [comment]
Sorry for that :/.

It's unclear to me why this error is suddenly occurring but I don't have
the time to investigate it further at the moment.

**Comment by brollb** [comment]
@le-cds do you see it in release 0.4.1 of elk? I built elkjs off of that release and seem to still see the issue... I tried building off of master but I am running into some issues.

**Comment by le-cds** [comment]
> @le-cds do you see it in release 0.4.1 of elk?

The image I posted was produced with the current nightly. I just tried again with our 0.4.1 code base and I'm getting the same (correct) result. I'm inclined to agree with @uruuru.

Is there a way for you to look at the actual coordinates and bend points produced by layout?

**Comment by uruuru** [comment]
You can look at the result by stepping through the code in a browser's
debugger.

I just tested the extended edge format: it seems to produce correct results
(as opposed to primitive edges).

**Comment by brollb** [comment]
After doing some more digging, it looks like the issue was in my visualization. I didn't realize that a section could have multiple bendpoints.... I was assuming that each vertical or horizontal edge would be a section (each with only a start and end point). I updated my visualizer and it all looks good now!

Also, to test the layout without any visualizer, I made a little cli script which takes a graph json (in a file) and prints the result to stdout. It is a pretty tiny amount of code but I thought it might be useful so I figured I would push it to github in case anyone else runs into a situation like this: https://github.com/brollb/elkjs-cli

It is worth noting that I am using elkjs built from v0.4.1 of elk in that repository.

Thanks again for all the help!
